### ISSUE TITLE
Manually include or bundle a js file as require/import on some other function than require / import .

### ISSUE THREAD

**Issue by redoC-A2k** [webpack/webpack#18446]
Suppose in my js code I have a function like 
createClient
 which takes a path to a JS FIle . I want that when webpack runs and find createclient from "sync-rpc", it does not use the createClient functionality from sync-rpc rather it reads the js file and assign the default exported functions from that given file to createClient to  getExtractedSummary , openAICall , callWikipediaApi .
I don't want to change source code so , I can't instead of createClient use require/import at that place . I want only that when webpack bundles or compile the js file and when it encouters createClient , It manually read the file and return the default exported function 
In other words I want some way to polyfill createClient funtion from sync-rpc library

**Comment by alexander-akait** [comment]
Please don't duplicate, close in favor of https://github.com/webpack/webpack/discussions/18447
